proto/rpc/v1/ledger_objects.proto

```diff
@@ -0,0 +1,141 @@
+syntax = "proto3";
+
+package rpc.v1;
+
+import "rpc/v1/amount.proto";
+
+message LedgerObject {
+
+    oneof object {
+        AccountRoot account_root = 1;
+        RippleState ripple_state = 2;
+        Offer offer = 3;
+        SignerList signer_list = 4;
+        DirectoryNode directory_node = 5;
+    }   
+}
+
+message DirectoryNode {
+
+    uint32 flags = 1;
+
+    bytes root_index = 2;
+
+    repeated bytes indexes = 3;
+
+    uint64 index_next = 4;
+
+    uint64 index_previous = 5;
+
+    string owner = 6;
+
+    bytes taker_pays_currency = 7;
+
+    bytes taker_pays_issuer = 8;
+
+    bytes taker_gets_currency = 9;
+
+    bytes taker_gets_issuer = 10;
+}
+
+message SignerList {
+
+    uint32 flags = 1;
+
+    bytes previous_txn_id = 2;
+
+    uint32 previous_txn_ledger_sequence = 3;
```

Comment:
No strong feelings, but it's really weird how the API sometimes abbreviates `transaction` to `txn`, but seems to never abbreviate anything else and it feels inconsistent to me. 

Would recommend thinking about whether this is worth renaming for consistency, but I ultimately don't feel that strongly.

Reply:
@keefertaylor Would you prefer expanding txn to transaction, or also abbreviating the other things (ledger to lgr and sequence to seq)?

Reply:
I expanded all abbreviations